rules_jvm_export/jvm_export/support/deploy.py

```diff
@@ -0,0 +1,264 @@
+                maven_url,
+                curl_opts,
+                sign(srcjar_path),
+                filename_base + "-sources.jar.asc",
+            )
+        # TODO(vmax): use real Javadoc instead of srcjar
+        upload(maven_url, curl_opts, srcjar_path, filename_base + "-javadoc.jar")
+        if args.gpg:
+            upload(
+                maven_url,
+                curl_opts,
+                sign(srcjar_path),
+                filename_base + "-javadoc.jar.asc",
+            )
+
+    with tempfile.NamedTemporaryFile(mode="wt", delete=True) as pom_md5:
+        pom_md5.write(md5(pom_file_path))
+        pom_md5.flush()
+        upload(maven_url, curl_opts, pom_md5.name, filename_base + ".pom.md5")
+
+    with tempfile.NamedTemporaryFile(mode="wt", delete=True) as pom_sha1:
+        pom_sha1.write(sha1(pom_file_path))
+        pom_sha1.flush()
+        upload(maven_url, curl_opts, pom_sha1.name, filename_base + ".pom.sha1")
+
+    with tempfile.NamedTemporaryFile(mode="wt", delete=True) as jar_md5:
+        jar_md5.write(md5(jar_path))
+        jar_md5.flush()
+        upload(maven_url, curl_opts, jar_md5.name, filename_base + ".jar.md5")
+
+    with tempfile.NamedTemporaryFile(mode="wt", delete=True) as jar_sha1:
+        jar_sha1.write(sha1(jar_path))
+        jar_sha1.flush()
+        upload(maven_url, curl_opts, jar_sha1.name, filename_base + ".jar.sha1")
+
+    if os.path.exists(srcjar_path):
+        with tempfile.NamedTemporaryFile(mode="wt", delete=True) as srcjar_md5:
+            srcjar_md5.write(md5(srcjar_path))
+            srcjar_md5.flush()
+            upload(
+                maven_url,
+                curl_opts,
+                srcjar_md5.name,
+                filename_base + "-sources.jar.md5",
+            )
+            # TODO(vmax): use checksum of real Javadoc instead of srcjar
+            upload(
+                maven_url,
+                curl_opts,
+                srcjar_md5.name,
+                filename_base + "-javadoc.jar.md5",
+            )
+
+        with tempfile.NamedTemporaryFile(mode="wt", delete=True) as srcjar_sha1:
+            srcjar_sha1.write(sha1(srcjar_path))
+            srcjar_sha1.flush()
+            upload(
+                maven_url,
+                curl_opts,
+                srcjar_sha1.name,
+                filename_base + "-sources.jar.sha1",
+            )
+            # TODO(vmax): use checksum of real Javadoc instead of srcjar
+            upload(
+                maven_url,
+                curl_opts,
+                srcjar_sha1.name,
+                filename_base + "-javadoc.jar.sha1",
+            )
+
+
+def _parse_pom(pom, elem_name):
+    namespaces = {"ns0": "http://maven.apache.org/POM/4.0.0"}
+    elem = pom.find(elem_name, namespaces)
+    if elem is None or len(elem.text) == 0:
+        raise Exception(f"Could not get {elem_name} from pom.xml")
+    return elem.text
+
+
+def sha1(fn):
+    return hashlib.sha1(open(fn, "rb").read()).hexdigest()
```

Comment:
Will this not leak a file descriptor?
```suggestion
    with open(fn, "rb") as f:
      return hashlib.sha1(f.read()).hexdigest()
```